File: Lab 1.2/C++/StringLibrary/StringManipulator.cpp
#include "StringManipulator.h"
#include <algorithm>

// Конструктор за замовчуванням
StringManipulator::StringManipulator() : text("") {}

// Конструктор із параметром
StringManipulator::StringManipulator(const std::string& input) {
    text = input;
    std::transform(text.begin(), text.end(), text.begin(), ::toupper);
}

// Конструктор копіювання
StringManipulator::StringManipulator(const StringManipulator& other) {
    text = other.text;
}

// Конструктор переміщення
StringManipulator::StringManipulator(StringManipulator&& other) noexcept {
    text = std::move(other.text);
}

// Деструктор
StringManipulator::~StringManipulator() {}

// Гетери
std::string StringManipulator::GetText() const {
    return text;
}

size_t StringManipulator::GetLength() const {
    return text.length();
}

// Метод заміни символа
void StringManipulator::ReplaceChar(char oldChar, char newChar) {
    std::replace(text.begin(), text.end(), oldChar, newChar);
}

File: Lab 1.2/C#/StringLibrary/StringManipulator.cs
using System;

namespace StringLibrary
{
    public class StringManipulator
    {
        private string text;  // Приватне поле

        // Конструктор за замовчуванням
        public StringManipulator()
        {
            text = "";
        }

        // Конструктор із параметром
        public StringManipulator(string input)
        {
            text = input.ToUpper();
        }

        // Конструктор копіювання
        public StringManipulator(StringManipulator other) : this(other.text) {}

        // Властивість (читання значення)
        public string Text => text;

        // Метод обчислення довжини рядка (без урахування пробілів)
        public int GetLength(bool ignoreSpaces = false)
        {
            return ignoreSpaces ? text.Replace(" ", "").Length : text.Length;
        }

        // Метод заміни символа (нечутливий до регістру)
        public void ReplaceChar(char oldChar, char newChar)
        {
            text = text.Replace(char.ToUpper(oldChar), char.ToUpper(newChar))
                       .Replace(char.ToLower(oldChar), char.ToLower(newChar));
        }
    }
}